readthedocs/search/documents.py

```diff
@@ -96,11 +96,13 @@ def update(self, thing, refresh=None, action='index', **kwargs):
         # TODO: remove this overwrite when the issue has been fixed
         # See below link for more information
         # https://github.com/sabricot/django-elasticsearch-dsl/issues/111
-        if isinstance(thing, HTMLFile):
+        # Moreover, do not need to check if its a delete action
+        # Because while delete action, the object is already remove from database
+        if isinstance(thing, HTMLFile) and action != 'delete':
```

Comment:
I did not consider delete case. While a object gets deleted, it calls `update` with `delete` action.